pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
```

Comment:
We can, but then we can't use versioner for it. And since we will need to use meta.Accessor, then there is no need to use versioner at all. WDYT?